test/widgets/message_list_test.dart

```diff
@@ -222,6 +238,87 @@ void main() {
     check(backgroundColor()).isSameColorAs(MessageListTheme.dark.streamMessageBgDefault);
   });
 
+  group('fetch initial batch of messages', () {
+    group('topic permalink', () {
+      final someStream = eg.stream();
+      const someTopic = 'some topic';
+
+      final otherStream = eg.stream();
+      const otherTopic = 'other topic';
+
+      testWidgets('with message move', (tester) async {
+        final narrow = TopicNarrow(someStream.streamId, eg.t(someTopic), with_: 1);
+        await setupMessageListPage(tester,
+          narrow: narrow,
+          // server sends the /with/<id> message in its current, different location
+          messages: [eg.streamMessage(id: 1, stream: otherStream, topic: otherTopic)],
+          streams: [someStream, otherStream],
+          subscriptions: [eg.subscription(someStream), eg.subscription(otherStream)],
```

Comment:
nit: I think the subscriptions aren't needed here

(some other tests need them because they use CombinedFeedNarrow)